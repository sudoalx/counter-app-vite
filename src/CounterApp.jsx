import { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  /*
    Tarea: Manejar butones de resta y reset con hooks
  */

  const handleSubstract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <div className="counter-display">
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
      </div>
      <div className="button-controls">
        <button aria-label="btn-minus" onClick={() => handleSubstract()}>
          -1
        </button>
        <button aria-label="btn-reset" onClick={() => handleReset()}>
          Reset
        </button>
        <button aria-label="btn-plus" onClick={() => handleAdd()}>
          +1
        </button>
      </div>
    </>
  );
};

CounterApp.defaultProps = {
  value: 0,
};

CounterApp.defaultProps = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
