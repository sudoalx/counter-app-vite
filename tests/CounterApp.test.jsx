import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";

const initialValue = 100;
describe("Testing <CounterApp/> component", () => {
  test("should match the snapshot", () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });
  test("should start the counter from val 100", () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toBe("100");
  });
  test("should increment the counter +1", () => {
    render(<CounterApp value={initialValue} />);
    const plusBtn = screen.getByRole("button", { name: "btn-plus" });
    fireEvent.click(plusBtn); // Se usa fireEvent() para accionar eventos como "click"
    expect(screen.getByText("101")).toBeTruthy();
  });
  test("should decrement the counter -1", () => {
    render(<CounterApp value={initialValue} />);
    const minusBtn = screen.getByRole("button", { name: "btn-minus" });
    fireEvent.click(minusBtn);
    expect(screen.getByText("99")).toBeTruthy();
  });
  test("should reset the counter back to zero", () => {
    render(<CounterApp value={initialValue} />);
    const resetBtn = screen.getByRole("button", { name: "btn-reset" }); // The name attr will be the aria-label defined in the JSX
    const plusBtn = screen.getByRole("button", { name: "btn-plus" });
    fireEvent.click(plusBtn);
    fireEvent.click(plusBtn);
    fireEvent.click(plusBtn);
    fireEvent.click(resetBtn);
    expect(screen.getByText(100)).toBeTruthy();
  });
});
