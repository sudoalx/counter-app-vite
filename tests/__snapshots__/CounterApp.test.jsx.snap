// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CounterApp/> component should match the snapshot 1`] = `
<div>
  <div
    class="counter-display"
  >
    <h1>
      CounterApp
    </h1>
    <h2>
      100
    </h2>
  </div>
  <div
    class="button-controls"
  >
    <button
      aria-label="btn-minus"
    >
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
    <button
      aria-label="btn-plus"
    >
      +1
    </button>
  </div>
</div>
`;
